lib/bmp280/calibration/bme680_calibration.ex

```diff
@@ -40,7 +40,8 @@ defmodule BMP280.BME680Calibration do
         <<par_h2h, par_h2l::4, par_h1l::4, par_h1h, par_h3::signed, par_h4::signed,
           par_h5::signed, par_h6, par_h7::signed, par_t1::little-16, par_gh2::little-signed-16,
           par_gh1::signed, par_gh3::signed>>,
-        <<res_heat_val, _skip01, _::2, res_heat_range::2, _::4, _skip03, range_switching_error>>
+        <<_::2, res_heat_val::signed-2, _::4, _skip01, _::2, res_heat_range::2, _::4, _skip03,
+          range_switching_error::signed-4, _::4>>
```

Comment:
The original bitstring matches the datasheet:

<img width="938" alt="image" src="https://user-images.githubusercontent.com/64669/112075456-e9bb2c80-8b4e-11eb-90d4-9c33aa574c93.png">

I.e. `res_heat_val` is the first byte and `res_head_range` is bits 5:4.

Where did you see that the bytes were swapped?

Reply:
I saw a bit mask is applied when data is parsed. I might misunderstand that part.
https://github.com/BoschSensortec/BME68x-Sensor-API/blob/90aebea28c7c685db97b7db41b07ec78948bbf3b/bme68x_defs.h#L555

Reply:
From those:

```c
        dev->calib.res_heat_range = ((coeff_array[BME68X_IDX_RES_HEAT_RANGE] & BME68X_RHRANGE_MSK) / 16);
        dev->calib.res_heat_val = (int8_t)coeff_array[BME68X_IDX_RES_HEAT_VAL];
        dev->calib.range_sw_err = ((int8_t)(coeff_array[BME68X_IDX_RANGE_SW_ERR] & BME68X_RSERROR_MSK)) / 16;
```

Here's what I see. 

`res_heat_val` is byte BME68X_IDX_RES_HEAT_VAL (37) in `coeff_array` and it's a signed byte. That's a bug in the original code since it was unsigned in the bitstring. 

`res_heat_range` is in byte BME68X_IDX_RES_HEAT_RANGE (39) in `coeff_array` and it's the bits in BME68X_RHRANGE_MSK (0x30) which are bits 5:4.  The C interprets it as unsigned and this matches the original bitstring. (You didn't change this)

I'm not sure if this was an issue with the URL, but the first one pointed to `BME68X_RUN_GAS_MSK` and that's not used in the C code.

So, I think there's a bug in that line of code and that it should be changed to `<<res_heat_val::signed, _skip01, _::2, res_heat_range::2, _::4, _skip03, range_switching_error>>`. 

I wonder how that changes things for you.

Reply:
Oh I see I totally mixed up `res_heat_range` and `res_heat_val`. Thanks!
Lastly I'd like to confirm `range_sw_err`.

```c
#define BME68X_RSERROR_MSK                        UINT8_C(0xf0)

dev->calib.range_sw_err = ((int8_t)(coeff_array[BME68X_IDX_RANGE_SW_ERR] & BME68X_RSERROR_MSK)) / 16;
```

Looks like only high 4 bits are taken. Also isn't it signed?

Disclaimer: I am unfamiliar with C and those "strict" data types

Reply:
Regarding `range_sw_err`, I didn't notice you changed that, but now I see. What you changed it to looks right. I.e. high 4-bits and interpreted as a signed integer.

Reply:
Cool! I guess this code should be good now.

Reply:
Yes. I marked as approved.